### ISSUE TITLE
[heft] NODE_ENV env variable not set during test phase

### ISSUE THREAD

**Issue by anthony-legay** [microsoft/rushstack#2420]
## Summary

So, the issue is pretty simple: Jest, as specified in its documentation should set 2 environment variables during its execution:
`JEST_WORKER_ID` and `NODE_ENV` (to value `test`).

Using `heft test` command which then calls Jest, it appears the `NODE_ENV` variable is not set, only `JEST_WORKER_ID`.
This breaks any test specific logic which uses this env variable.

As discussed with @octogonz:
>However personally, I would prefer NOT to rely on something as brittle as an environment variable. It would be better to declare an API like if (TEST) { } or if (jestEnv.TEST) { } that can have proper TypeScript typings. My team was recently asked to set up something similar for passing down Webpack variables, so maybe these can be combined into a single API.

**Comment by octogonz** [comment]
Thanks for reporting this. I'll be back from vacation next week and can take a look.

Supporting the environment variable is relatively easy (and maybe could be done as a first PR to get you unblocked).

But we should work out the "right way" to pass down environment settings when using Heft.  Seems like Webpack variables and Jest variables should be combined into a single global API.

**Comment by octogonz** [comment]
Let's use this issue to track the "easy" fix of exposing `NODE_ENV` to Heft's Jest environment.  It is the title of this issue.  If we want to provide a better way, we can create a separate issue to track that idea (or just make a PR).

@Lebowskovitch are you able to make a PR?  Heft's Jest config is set up [here](https://github.com/microsoft/rushstack/blob/f1e082967021baa3efb69968f193303086aa55c3/apps/heft/src/plugins/JestPlugin/JestPlugin.ts#L58).

**Comment by woss** [comment]
Since heft-jest-plugin(HJP) is using the `@jest/core` the `NODE_ENV` is not set at all because jest sets it in the CLI and REPL scripts.

One way is to add the new optional field `setNodeEnvVar` to `IJestPluginOptions` where the default value would be `true`.  If unchanged this would behave precisely as jest implementation:

> NODE_ENV
> Set to 'test' if it's not already set to something else.

If you agree with this I'll make a PR and add it, this would save me a lot of work of setting up the NODE_ENV for over 40 pacakges

Screenshot from the jest repo:
<img width="737" alt="image" src="https://github.com/microsoft/rushstack/assets/80440/d108ba5d-5f98-43f5-959e-3fffe1ecac00">
